finagle-http/src/main/scala/com/twitter/finagle/http/Status.scala

```diff
@@ -29,7 +29,54 @@ case class Status(code: Int) {
     }
 }
 
+
 object Status {
+
+  /**
+   * Matches when the status code isn't in range of any of the  categories: {{Informational}},
+   * {{Successful}}, {{Redirection}}, {{ClientError}}, {{ServerError}}.
+   */
+  object UnknownStatus {
+    def unapply(status: Status): Option[Status] =
+      Option(status).filter(s => s.code < 100 || s.code >= 600)
```

Comment:
This should match line 77 to be consistent. I prefer ```Option```, but I may be the only one so consistency is more important.